lib/compress/zstd_compress.c

```diff
@@ -6029,10 +6078,15 @@ ZSTD_compressSequences_internal(ZSTD_CCtx* cctx,
     }
 
     while (remaining) {
+        size_t compressedSeqsSize;
         size_t cBlockSize;
         size_t additionalByteAdjustment;
-        lastBlock = remaining <= cctx->blockSize;
-        blockSize = lastBlock ? (U32)remaining : (U32)cctx->blockSize;
+        size_t blockSize = determine_blockSize(cctx->appliedParams.blockDelimiters,
+                                        cctx->blockSize, remaining,
+                                        inSeqs, inSeqsSize, seqPos);
+        U32 const lastBlock = (blockSize == remaining);
+        assert(blockSize <= remaining);
```

Comment:
Looks like this assert is failing.

Reply:
Yes, that's exactly the issue, 
`blockSize` can be an error code, so this `assert()` is only valid after the `FORWARD_IF_ERROR()`.